cannon/mipsevm/singlethreaded/instrumented.go

```diff
@@ -47,20 +37,14 @@ func NewInstrumentedState(state *State, po mipsevm.PreimageOracle, stdOut, stdEr
 	}
 }
 
-func NewInstrumentedStateFromFile(stateFile string, po mipsevm.PreimageOracle, stdOut, stdErr io.Writer, meta *program.Metadata) (*InstrumentedState, error) {
-	state, err := serialize.Load[State](stateFile)
-	if err != nil {
-		return nil, err
-	}
-	return NewInstrumentedState(state, po, stdOut, stdErr, meta), nil
-}
-
-func (m *InstrumentedState) InitDebug() error {
-	stackTracker, err := exec.NewStackTracker(m.state, m.meta)
+func (m *InstrumentedState) InitDebug(meta mipsevm.Metadata) error {
+	stackTracker, err := exec.NewStackTracker(m.state, meta)
 	if err != nil {
 		return err
 	}
+	m.meta = meta
 	m.stackTracker = stackTracker
+	m.sleepCheck = meta.CreateSymbolMatcher("runtime.notesleep")
```

Comment:
Oh good spot - the old code would use a no-op sleepCheck if meta was nil but meta was actually always non-nil.  I've switched things round so that meta is always passed into `CreateVM` (for single and multithreaded) and not into `InitDebug` so if you provide a metadata file the sleep check will be active.  And then the symbol look up behaviour will be consistent between single and multithreaded too.